### PR TITLE
Update tooling to use Eslint and Rollup

### PR DESCRIPTION
**Pull request by TerrenceLJones**
Welcome to the future.

### CODE REVIEW

**Comment by TerrenceLJones** [comment]
This resolves #140

File: package.json [hunk 1]
```diff
@@ -34,21 +34,24 @@
...
     "babel": "^5.8.19",
+    "babel-eslint": "^6.0.2",
+    "babel-preset-es2015-rollup": "^1.1.1",
     "babelify": "^6.3.0",
     "bluebird": "^2.9.27",
     "browserify": "^8.1.1",
     "chai": "^2.0.0",
     "del": "^1.1.1",
+    "eslint": "^2.8.0",
     "esperanto": "^0.6.7",
```
Comment (paulfalgout) [inline package.json]: need to axe the unused dependencies

Comment (TerrenceLJones) [inline package.json]: smh...yep...I'm going to force push some changes.

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  "rules": {
+    "array-bracket-spacing": [2, "never"],
+    "arrow-body-style": 2,
+    "block-scoped-var": 2,
+    "block-spacing": [2, "always"],
+    "brace-style": 2,
+    "camelcase": 2,
+    "comma-spacing": [2, {"before": false, "after": true}],
+    "comma-style": 2,
+    "complexity": [2, 11],  // FIXME: Ideally this could be reduced to 5
```
Comment (paulfalgout) [inline .eslintrc]: This probably can be reduced to 5 for this lib and we can axe the comment.

File: gulpfile.js [hunk 3]
```diff
@@ -1,21 +1,28 @@
-var gulp = require('gulp');
-var $ = require('gulp-load-plugins')();
-const fs = require('fs');
+const gulp = require('gulp');
+const $ = require('gulp-load-plugins')();
 const del = require('del');
-const glob = require('glob');
+const file = require('gulp-file');
```
Comment (paulfalgout) [inline gulpfile.js]: remove any `gulp-` and use the `$.file` :-( for now

File: package.json [hunk 4]
```diff
@@ -77,19 +78,11 @@
...
-      "stub",
-      "spy",
-      "expect",
-      "_",
-      "Backbone",
-      "Marionette"
-    ]
+    }
   },
   "dependencies": {
```
Comment (paulfalgout) [inline package.json]: change this to `peerDependencies` and revert the changes on versions for `backbone` and `backbone.marionette` below

File: .travis.yml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 language: node_js
 node_js:
-  - "0.10"
+  - "4"
```
Comment (paulfalgout) [inline .travis.yml]: we also need to 
```diff
- - UNDERSCORE=1.8 BACKBONE=1.2 MARIONETTE=2.4 MAINRUN=true
+ - UNDERSCORE=1.8 BACKBONE=1.3.2 MARIONETTE=2.4 MAINRUN=true
```

File: gulpfile.js [hunk 6]
```diff
@@ -30,107 +30,104 @@ gulp.task('clean-tmp', function(cb) {
...
-        return js_source.replace('./state-class', 'marionette.toolkit.state-class');
-      }
-
-      return js_source;
+function _generate(bundle) {
+  const intro = getBanner();
+
+  return bundle.generate({
+    format: 'umd',
+    moduleName: config.exportVarName,
```
Comment (paulfalgout) [inline gulpfile.js]: we should pass in the moduleName as for the packages it should be using the `exportVarName`

File: .travis.yml (hunk 5 above)
Comment (TerrenceLJones) [inline .travis.yml]: @paulfalgout For confirmation...you want me to replace the first line with the second? Or do you want me to just add the second line?

Comment (paulfalgout) [inline .travis.yml]: Yeah I made a diff comment so it should look like the change with the -  of one line for removal in red and the + in green for the add

**Comment by paulfalgout** [comment]
`gulp test-browser` isn't working.  After running it you should be able to run `runner.html`
